### PR TITLE
Add date header

### LINKED ISSUE #20: Set date header
https://www.mail-tester.com/ reports that emails sent from mailyak are missing a date header, which penalizes the spam score.

Is there currently a way to set the date header, or is such a feature planned?

Many thanks.

### PR DESCRIPTION
**Pull request by rwhittington**
Fixes #20 
Adds current Date/Time to MIME header

### CODE REVIEW

**Comment by rwhittington** [comment]
Apologies, didn't run all tests on my local machine before putting in the pull request.
Removed broken pull request, and re-submitted.

**Comment by domodwyer** [comment]
Hi @rwhittington

**Thanks!** I've been out the country so didn't get a chance to work on this - I really appreciate it!

It's probably best to set the `MailYak.date` in `New()` to the current value though - a process that's been running for a year (like an API server) would be sending emails with `Date: <one year ago>` otherwise!

Again, I really appreciate the PR, thanks very much!

Dom

File: mailyak.go [hunk 1]
```diff
@@ -47,6 +48,7 @@ func New(host string, auth smtp.Auth) *MailYak {
 		auth:           auth,
 		trimRegex:      regexp.MustCompile("\r?\n"),
 		writeBccHeader: false,
+		date:           currentDateTime,
```
Comment (domodwyer) [inline mailyak.go]: This should be set to `time.Now().Format(time.RFC1123Z)` for each call, rather than as a global.

**Comment by rwhittington** [comment]
@domodwyer No problem, happy to help if I can! 

Thanks for pointing that out! I've pushed the changes so New() generates a date/time every time.

I've kept the global variable in, so all the tests run against one value.

Ryan

**Comment by domodwyer** [comment]
Fantastic! Thanks @rwhittington - it's much appreciated.

Dom

**Pull request merged**